librz/core/cbin.c

```diff
@@ -1043,10 +1065,367 @@ RZ_API bool rz_core_bin_apply_relocs(RzCore *core, RzBinFile *binfile, bool va_b
+	} else if (info->bits == 16 && bits == 32) {
+		force_bits = 32;
+	} else if (!(paddr & 1) && bits == 32) {
+		force_bits = 32;
+	}
+	if (force_bits) {
+		rz_analysis_hint_set_bits(core->analysis, addr, force_bits);
+	}
+}
+
+static void handle_arm_symbol(RzCore *core, RzBinObject *o, RzBinSymbol *symbol, int va) {
+	handle_arm_hint(core, o, symbol->paddr, symbol->vaddr, symbol->bits, va);
+}
+
+static void handle_arm_entry(RzCore *core, RzBinObject *o, RzBinAddr *entry, int va) {
+	handle_arm_hint(core, o, entry->paddr, entry->vaddr, entry->bits, va);
+}
+
+static void select_flag_space(RzCore *core, RzBinSymbol *symbol) {
+	if (symbol->is_imported) {
+		rz_flag_space_push(core->flags, RZ_FLAGS_FS_IMPORTS);
+	} else if (symbol->type && !strcmp(symbol->type, RZ_BIN_TYPE_SECTION_STR)) {
+		rz_flag_space_push(core->flags, RZ_FLAGS_FS_SYMBOLS_SECTIONS);
+	} else {
+		rz_flag_space_push(core->flags, RZ_FLAGS_FS_SYMBOLS);
+	}
+}
+
+RZ_API bool rz_core_bin_apply_symbols(RzCore *core, RzBinFile *binfile, bool va) {
+	rz_return_val_if_fail(core && binfile, NULL);
+	RzBinObject *o = binfile->o;
+	RzBinInfo *info = o ? o->info : NULL;
+	if (!info) {
+		return false;
+	}
+
+	bool is_arm = info && info->arch && !strncmp(info->arch, "arm", 3);
+	bool bin_demangle = rz_config_get_b(core->config, "bin.demangle");
+	const char *lang = bin_demangle ? rz_config_get(core->config, "bin.lang") : NULL;
+
+	rz_spaces_push(&core->analysis->meta_spaces, "bin");
+	rz_flag_space_set(core->flags, RZ_FLAGS_FS_SYMBOLS);
+
+	RzList *symbols = rz_bin_get_symbols(core->bin);
+	size_t count = 0;
+	RzListIter *iter;
+	RzBinSymbol *symbol;
+	rz_list_foreach (symbols, iter, symbol) {
+		if (!symbol->name) {
+			continue;
+		}
+		ut64 addr = rva(o, symbol->paddr, symbol->vaddr, va);
+		SymName sn = { 0 };
+		count++;
+		sym_name_init(core, &sn, symbol, lang);
+		char *rz_symbol_name = rz_str_escape_utf8(sn.name, false, true);
+
+		if (is_section_symbol(symbol) || is_file_symbol(symbol)) {
+			/*
+			 * Skip section symbols because they will have their own flag.
+			 * Skip also file symbols because not useful for now.
+			 */
+		} else if (is_special_symbol(symbol)) {
+			if (is_arm) {
+				handle_arm_special_symbol(core, o, symbol, va);
+			}
+		} else {
+			// TODO: provide separate API in RzBinPlugin to let plugins handle analysis hints/metadata
+			if (is_arm) {
+				handle_arm_symbol(core, o, symbol, va);
+			}
+			select_flag_space(core, symbol);
+			/* If that's a Classed symbol (method or so) */
+			if (sn.classname) {
+				RzFlagItem *fi = rz_flag_get(core->flags, sn.methflag);
+				if (core->bin->prefix) {
+					char *prname = rz_str_newf("%s.%s", core->bin->prefix, sn.methflag);
+					rz_name_filter(sn.methflag, -1, true);
+					free(sn.methflag);
+					sn.methflag = prname;
+				}
+				if (fi) {
+					rz_flag_item_set_realname(fi, sn.methname);
+					if ((fi->offset - core->flags->base) == addr) {
+						//		char *comment = fi->comment ? strdup (fi->comment) : NULL;
```

Comment:
Is this comment necessary? Maybe better remove it?

Reply:
comments are never necessary, removing.